python/src/equistore/labels.py

```diff
@@ -271,25 +271,81 @@ def arange(name, *args):
 
         :returns: A new :py:class:`Labels` object with the provided name and
             values corresponding to the specified range bounds and step.
+
+        Here is an example using this method:
+
+        >>> from equistore import Labels
+        ...
+        >>> # Construct a labels object using Labels.arange():
+        >>> labels = Labels.arange("dummy", 7)
+        >>> # Inspect the labels object:
+        >>> print(labels.names)
+        ('dummy',)
+        >>> print(labels)
+        [(0,) (1,) (2,) (3,) (4,) (5,) (6,)]
+        >>> # The same can be accomplished by specifying the names for
+        >>> # the integer arguments. For example:
+        >>> labels = Labels.arange("dummy", start=2, stop=6)
+        >>> print(labels)
+        [(2,) (3,) (4,) (5,)]
+        >>> # However, a mixture of the two will not work:
+        >>> labels = Labels.arange("dummy", 2, stop=6)
+        Traceback (most recent call last):
+            ...
+        ValueError: please provide either no names or all names for the integer \
+arguments to `Labels.arange()`, but not a mixture of the two
         """
 
         args_len = len(args)
-        if args_len == 0:
+        kwargs_len = len(kwargs)
+
+        if args_len == 0 and kwargs_len == 0:
             raise ValueError(
                 "please provide at least one integer for `Labels.arange()`"
             )
-        if args_len > 3:
+        if args_len != 0 and kwargs_len != 0:
             raise ValueError(
-                f"the maximum number of integer arguments accepted by \
-                `Labels.arange()` is 3. {args_len} were provided"
+                "please provide either no names or all names for the integer "
+                "arguments to `Labels.arange()`, but not a mixture of the two"
             )
-        for arg in args:
-            if not np.issubdtype(type(arg), np.integer):
+        if args_len == 0:
+            has_args = False
+        else:
+            has_args = True
+
+        if has_args:
+            if args_len > 3:
                 raise ValueError(
-                    "all numbers provided to `Labels.arange()` must be integers"
+                    "the maximum number of integer arguments accepted by "
+                    f"`Labels.arange()` is 3. {args_len} were provided"
                 )
-
-        labels = Labels(names=[name], values=np.arange(*args).reshape(-1, 1))
+            for arg in args:
+                if not np.issubdtype(type(arg), np.integer):
+                    raise ValueError(
+                        "all numbers provided to `Labels.arange()` must be integers"
+                    )
```

Comment:
Do we need to check this ourself? Shouldn't numpy do this for us?

Reply:
No, unfortunately `np.arange` also works with floats, and it gives float ranges out in that case, which would not work for us. It works a bit like linspace with slightly different arguments in that case

Reply:
Ok, perfect. I've added the comments